src/Lumper.UI/ViewModels/Shared/Pakfile/PakfileLumpViewModel.cs

```diff
@@ -210,86 +214,76 @@ PakfileEntryTextViewModel entry in Entries
             if (entry.Content is null)
                 return;
 
-            string updatedOp = oldPath;
-            string updatedNp = newPath;
-            int matchIndex = -1;
             int changes = 0;
 
+            // Base index that tracks progress iterating through entire file
+            int baseIdx = -1;
             bool tryWithoutExtension = IgnorableExtensions.TryGetValue(
                 Path.GetExtension(entry.Key),
                 out string[]? opNoExtension
             );
+
+            // Outer loop as we traverse the entire file - could have multiple matches.
             while (true)
             {
-                int match = entry.Content.IndexOf(opNoPrefix, startIndex: matchIndex + 1, cmp);
+                string op = oldPath;
+                string np = newPath;
+
+                // If we matched a directory, we're almost certainly right to omit it unless the pakfile is completely
+                // fucked.
+                if (directoryMatch is not null)
+                {
+                    op = string.Join("/", op.Split('/')[1..]);
```

Comment:
It might be more efficient to use substrings:
`op = op.Substring(op.IndexOf("/") + 1);`

Reply:
Switched to using `Substring` where applicable in new version of pakfile refactoring